### PR TITLE
[eclipse/xtext#1569] Refactor Xtend to Java

### PR DESCRIPTION
**Pull request by kthoms**
For bundle org.eclipse.xtext.builder.standalone, except class
StandaloneBuilder.

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.builder.standalone/.settings/org.eclipse.jdt.ui.prefs [hunk 1]
```diff
@@ -1,54 +1,65 @@
-#Wed Oct 14 22:36:24 CEST 2009
```
Comment (tivervac) [inline org.eclipse.xtext.builder.standalone/.settings/org.eclipse.jdt.ui.prefs]: Good to see you enable these, but shouldn't these be copy pasted to every xtext plugin?

File: org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/compiler/CompilerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * @author Dennis Huebner - Initial contribution and API
+ */
+public class CompilerConfiguration {
+	private String sourceLevel = "1.5";
+
+	private String targetLevel = "1.5";
+
+	private boolean verbose;
```
Comment (tivervac) [inline org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/compiler/CompilerConfiguration.java]: Not sure if there's a policy for xtext, but we like to explicitly set the initialized value on non final primitive fields

File: org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollector.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+			LOG.debug("Finished collecting source models. Took: " + (System.currentTimeMillis() - startedAt) + " ms.");
+		}
+		return result;
+	}
+
+	protected Set<URI> collectResources(Iterable<URI> roots, Set<String> fileExtensions) {
+		Set<String> extensions = new HashSet<>(fileExtensions);
+		extensions.add("java");
+
+		List<String> paths = stream(roots.spliterator(), false).map(uri -> uri.toFileString()).collect(toList());
```
Comment (tivervac) [inline org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollector.java]: Can't we use `map(URI::toFileString)` here?

File: org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollector.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			}
+		});
+
+		return IterableExtensions.toSet(modelsFound.values());
+	}
+
+	protected void registerBundle(File file) {
+		try (JarFile jarFile = new JarFile(file)){
+			Manifest manifest = jarFile.getManifest();
+			if ((manifest == null)) {
```
Comment (tivervac) [inline org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollector.java]: Double parentheses

File: org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollector.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+				return;
+			}
+			String name = manifest.getMainAttributes().getValue("Bundle-SymbolicName");
+			if ((name != null)) {
```
Comment (tivervac) [inline org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollector.java]: Double parentheses

File: org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollector.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+				int indexOf = name.indexOf(";");
+				if ((indexOf > 0)) {
```
Comment (tivervac) [inline org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollector.java]: Double parentheses

File: org.eclipse.xtext.builder.standalone/.settings/org.eclipse.jdt.ui.prefs (hunk 1 above)
Comment (kthoms) [reply org.eclipse.xtext.builder.standalone/.settings/org.eclipse.jdt.ui.prefs]: removed the file change from this change

File: org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/compiler/CompilerConfiguration.java (hunk 2 above)
Comment (kthoms) [reply org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/compiler/CompilerConfiguration.java]: I don't see a value in doing this for booleans.

File: org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollector.java (hunk 3 above)
Comment (kthoms) [reply org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollector.java]: yes; I'll also add a test case

File: org.eclipse.xtext.builder.standalone.tests/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollectorTest.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import org.eclipse.emf.common.util.URI;
+import org.junit.Test;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+public class ResourceURICollectorTest {
+
+	@Test
+	public void test_collectAllResources () {
```
Comment (tivervac) [inline org.eclipse.xtext.builder.standalone.tests/src/org/eclipse/xtext/builder/standalone/incremental/ResourceURICollectorTest.java]: Very weird naming convention

**Pull request merged**